metrics/config.go

```
@@ -134,28 +167,30 @@ func UpdateExporterFromConfigMap(domain string, component string, logger *zap.Su
 			return
 		}
 
-		if isMetricsConfigChanged(newConfig) {
-			if err := newMetricsExporter(newConfig, logger); err != nil {
+		if isNewExporterRequired(newConfig) {
```

Should this function return an error?

---

I don't see any case it could return an error.

---

well you log an error. If it's not an error, then logging it as an error, also doesn't make sense?

---

The error is returned from `newMetricsExporter`. I think you were asking whether `isNewExporterRequired` should return an error?